src/saveload/afterload.cpp

```diff
@@ -1024,6 +1024,12 @@ bool AfterLoadGame()
 		_settings_game.construction.freeform_edges = false;
 	}
 
+	/* Force water borders to none for old savegames, because we used to allow terraforming edge tiles, and void tiles could be affected.
+	 * This prevents incorrectly drawing infinite water. */
+	if (IsSavegameVersionBefore(SLV_INFINITE_WATER_BORDERS)) {
+		_settings_game.game_creation.water_borders = 0;
```

Comment:
Why not just call `CheckWaterBorders()` instead? As we can detect whether water-border are still a fact, that seems a better transition for most games? :)

Reply:
I don't want to force new behavior on old savegames, to minimize pitchforks.

Reply:
I think that is the wrong call here. You do force it on new games (and it cant be disabled), so you only create more confusion this way. I would rather we have this as feature: if you have all water, it shows all water. Easier to explain, easier to get used to :)

Be bold! :D

Reply:
If you really anticipate pitchforks then it should be user-toggleable in any case, otherwise don't make it weirdly conditional on something the player can't properly control.